src/box/memtx_tree.cc

```diff
@@ -1457,6 +1600,26 @@ memtx_tree_index_create_iterator_after(struct index *base,
 	if (USE_HINT)
 		it->current.set_hint(HINT_NONE);
 	it->current_func_key = NULL;
+	if (after != NULL) {
+		it->after_data.key = after;
+		it->after_data.part_count = cmp_def->part_count;
+		if (USE_HINT)
+			it->after_data.set_hint(HINT_NONE);
+		if (it->type != ITER_EQ && it->type != ITER_REQ) {
+			/*
+			 * If we want to start after specified position with
+			 * range iterator, we need to turn iterator with
+			 * non-strict inequality into its analogue with
+			 * strict inequality.
+			 */
+			it->type = iterator_type_is_reverse(it->type) ?
+						 ITER_LT :
+						 ITER_GT;
```

Comment:
I don't like that we overwrite `it->type` for some iterator types (le, ge), but not for others (eq, req). Can you avoid that?

Reply:
We can overwrite `GE` and 'LE` inside `tree_iterator_start` or move this logic to https://github.com/tarantool/tarantool/blob/dc147ec9190fc8bbbe56ce8a4a1718fbf8b03484/src/box/memtx_tree.cc#L681-L689

Reply:
Right, I think we'd better do that in `tree_iterator_start` - this way `tree_iterator->key` and `tree_iterator->type` would denote the original iteration key and direction.

Reply:
Moved changing range iterators with equality to `tree_iterator_start`, `tree_iterator->key` isn't affected anymore.